From ad8bfb0f7ec6118ff000c474e0f2a31de9101b7b Mon Sep 17 00:00:00 2001
From: Ma Wupeng <mawupeng1@huawei.com>
Date: Mon, 22 May 2023 16:41:50 +0800
Subject: [PATCH 040/263] [Huawei] mm: mem_reliable: Fix reliable page counter
 mismatch problem

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I77BDW

--------------------------------

During copy_present_pte, rss counter is increased but the corresponding
reliable page counter is not updated. This will lead to reliable page
counter mismatch. Fix this by adding reliable page counter.

Fixes: 08966e4ea85c ("proc: Count reliable memory usage of reliable tasks")

Signed-off-by: Ma Wupeng <mawupeng1@huawei.com>
---
 mm/memory.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/memory.c b/mm/memory.c
index 32c2d5becb71..c4ecf8a23be8 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -887,6 +887,7 @@ copy_present_pte(struct vm_area_struct *dst_vma, struct vm_area_struct *src_vma,
 		get_page(page);
 		page_dup_rmap(page, false);
 		rss[mm_counter(page)]++;
+		reliable_page_counter(page, dst_vma->vm_mm, 1);
 	}
 
 	/*
-- 
2.17.1

